src/main/java/com/demo/classroom/DTO/RegistrationDTO.java

```diff
@@ -0,0 +1,27 @@
+package com.demo.classroom.DTO;
+import com.demo.classroom.Utility.Role;
+
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Size;
+import lombok.Data;
+
+@Data
+public class RegistrationDTO {
+    @NotBlank(message = "Username is mandatory")
+    @Size(min = 3, max = 20, message = "Username must be between 3 and 20 characters")
+    private String username;
+
+    @NotBlank(message = "Email is mandatory")
+    @Email(message = "Email should be valid")
+    private String email;
+
+    @NotBlank(message = "Password is mandatory")
+    @Size(min = 8, message = "Password must be at least 8 characters long")
+    private String password;
+
+    private Role role; 
```

Comment:
Solved. It is now strongly handled from the service layer. Anything from the frontend with `Role`is empty, null, or something else it will give response `Invalid Role Selected` with proper error message structure.

Below are the pictures: 
![Role-invalid-1](https://github.com/user-attachments/assets/b3d2970c-d6ed-455c-a50e-e9c512e25100)